cmd/cli/kubectl-kyverno/commands/apply/command.go

```diff
@@ -455,6 +466,18 @@ func (c *ApplyCommandConfig) initStoreAndClusterClient(store *store.Store, skipI
 			return nil, nil, skipInvalidPolicies, nil, nil, err
 		}
 	}
+	if len(vapParams) > 0 && !c.Cluster {
+		var fakeClientResources []runtime.Object
+		for _, p := range vapParams {
+			fakeClientResources = append(fakeClientResources, p)
+		}
+
+		dClient, err = dclient.NewFakeClient(runtime.NewScheme(), map[schema.GroupVersionResource]string{}, fakeClientResources...)
+		dClient.SetDiscovery(dclient.NewFakeDiscoveryClient(nil))
+		if err != nil {
+			return nil, nil, skipInvalidPolicies, nil, nil, err
+		}
+	}
```

Comment:
Why do we need to initiate a fake client?
In the case of offline mode, we will fetch the parameter resource from the file.

Reply:
My rationale behind doing so is that i wanna unify the logic. 
Instead of treating the offline operation as something different from the online one, its useful to think of it as just the same as the online op but the client is a fake client.

Let me know what you think